replica/database.cc

```diff
@@ -362,6 +363,15 @@ database::database(const db::config& cfg, database_config dbcfg, service::migrat
             utils::updateable_value(std::numeric_limits<uint32_t>::max()),
             utils::updateable_value(std::numeric_limits<uint32_t>::max()),
             reader_concurrency_semaphore::register_metrics::yes)
+    , _view_update_read_concurrency_sem(
+            utils::updateable_value<int>(max_count_concurrent_reads),
```

Comment:
Better to use a separate limit.

Reply:
I added a dedicated limit for this, for now it's the same as the one for the user read semaphore, but I'm not sure there's a good value for this, as the view update reads don't go through the admission that the regular writes do, and as such the count never increases over 0 anyway

Reply:
If the view reads do not go through admission, then all this patch achieves is remove the view update reads from the reader semaphore. They will be just as out-of-control as before.
Note that just because the resources used by these reads are not accounted on the user read semaphore, they are still resources that are consumed and the system will have that much less resources for other things.

Reply:
I was planning to fix this in a follow-up patch, but it probably better to just add it here - I did that in the rebase